pkg/pool-weighted/contracts/smart/ManagedPool.sol

```diff
@@ -231,7 +243,9 @@ contract ManagedPool is BaseWeightedPool, ReentrancyGuard {
         endWeights = new uint256[](totalTokens);
 
         for (uint256 i = 0; i < totalTokens; i++) {
-            endWeights[i] = _tokenState[tokens[i]].decodeUint32(_END_WEIGHT_OFFSET).uncompress32();
+            endWeights[i] = _normalizeWeight(
+                _tokenState[tokens[i]].decodeUint64(_END_DENORM_WEIGHT_OFFSET).uncompress64(_MAX_DENORM_WEIGHT_SUM)
+            );
```

Comment:
I would be useful to add a comment explaining why `uncompress64(_MAX_DENORM_WEIGHT_SUM)`